server/index.js

```diff
@@ -42,6 +42,29 @@ app.get('/api/products', (req, res, next) => {
     });
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const id = parseInt(req.params.productId, 10);
+
+  const sql = `
+  select *
+      from products
+      where "productId" = $1
+  `;
+
+  const params = [id];
+
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next();
```

Comment:
here you use the `ClientError`

> - The requested product does not exist (call next() with a new ClientError())

`next(new ClientError('something helpful and informative about not finding the product'))`